### ISSUE TITLE
React 16.2 fragment syntax triggers

### ISSUE THREAD

**Issue by remcohaszing** [babel/babel-eslint#554]
Reacr 16.2 introduced fragments. This feature is accompanied by a syntactic sugar. The following component is now valid in React:

```jsx
import React from 'react';


export default () => (
  <>
    <div />
  </>
)
```

When using the default ESLint parser, this causes a syntax error.
When using the `babel-eslint` parser, this triggers the quotes rule.

**Comment by sscaff1** [comment]
Any updates on this? I'm using the following versions:
"babel-eslint": "^8.0.2",
"eslint": "^4.10.0",
"eslint-plugin-react": "^7.5.1",

**Comment by jacobp100** [comment]
As a workaround, you can use `<React.Fragment>...</React.Fragment>`

**Comment by mattiascaricato** [comment]
+1

**Comment by cristian-sima** [comment]
Any update?

**Comment by cristian-sima** [comment]
You can try 
```
jsx

import React, { Fragment } from 'react';

export default () => (
  <Fragment>
    <div />
  <Fragment/>
)
```

**Comment by ThaJay** [comment]
Of course we can use `Fragment` component but that is not the point here. We want to use fragment shorthand.
https://reactjs.org/docs/react-api.html#reactfragment

**Comment by jimmybrawn** [comment]
Maybe you can use a workaround like, create a component named "f" and make it import fragment from react and pass down its prop children. This way you can use <f></f> not sure if this will work

**Comment by jimmybrawn** [comment]
Maybe you can use a workaround like, create a component named "f" and make it import fragment from react and pass down its prop children. This way you can use `<f></f>` not sure if this will work

**Comment by jimmybrawn** [comment]
Or you can just import fragment under the same "f" and alias 😂

**Comment by fabiomcosta** [comment]
Note: Fragment is only working for `babel-eslint@^8`, v7 doesn't support that, as wrongly stated on the React Fragment blog post.
eslint can be any bigger than 3
https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html#eslint

**Comment by graingert** [comment]
@fabiomcosta nope that still triggers the quotes rule

**Comment by ReeganExE** [comment]
```json
"babel-eslint": "^8.2.2",
"eslint": "^4.19.1"
```
The latest ones still have quotes error.

**Comment by existentialism** [comment]
Just an update, this issue isn't with `babel-eslint` itself, as there isn't really anything we need to do related to _parsing_, the `quotes` rule itself doesn't know how to handle `JSXFragment` node types.

Until it get updated in ESLint proper (can track via the linked thread above), I've added a [new `quotes` rule](https://github.com/babel/eslint-plugin-babel/pull/139) in `eslint-plugin-babel` that won't trip on JSX shorthand syntax.

**Comment by hzoo** [comment]
Published with https://github.com/babel/eslint-plugin-babel/releases/tag/v5.0.0

It has a peerDep on ESLint 4.

**Comment by brneto** [comment]
`"babel-eslint": "^8.2.2",
"eslint": "^4.19.1",
"eslint-plugin-babel": "^5.0.0"`

I'm using that versions above, but still showing the error:
`ERROR in ./src/client/index.js
Module build failed: Thread Loader (Worker 0)
SyntaxError: Unexpected token (16:3)`

**Comment by existentialism** [comment]
@brneto that is a different error, can you create another issue with your config/code? better yet, a small repo that reproduces the problem?

**Comment by brneto** [comment]
Forget it, I fixed the problem! It had nothing to do with babel-eslint. It had to do with babel itself. I was using babel v6. When I've updated to v7 plus others configurations, worked! 

Anyway thanks for the help!

**Comment by jaydenseric** [comment]
Still getting the `Strings must use singlequote` error with:

- `eslint@5.0.0`
- `babel-eslint@8.2.5`

Finding it very difficult to debug why.

**Comment by jimmybrawn** [comment]
@jaydenseric did u try with babel 9 as stated above?

**Comment by jaydenseric** [comment]
> did u try with **babel 9**

I think you mean Babel v7 beta?

It's a `next@6.0.4-canary.8` project, which has the dependency [`@babel/core@7.0.0-beta.42`](https://github.com/zeit/next.js/blob/6.0.4-canary.8/package.json#L57). I tried manually adding the latest Babel v7 to my project dependencies, but it didn't help.

**Comment by viperfx07** [comment]
> Still getting the `Strings must use singlequote` error with:
> 
> * `eslint@5.0.0`
> * `babel-eslint@8.2.5`
> 
> Finding it very difficult to debug why.

Same here with
`eslint@5.5.0`
`babel-eslint@8.2.6`

**Comment by yuritoledo** [comment]
any update?

**Comment by st-zelenin** [comment]
not reproduced with:
```
"babel-eslint": "~10.0.1",
"eslint": "~5.11.1",
```

**Comment by yuritoledo** [comment]
the package that comes with Create React App, `react-app` fix it.

**Comment by johnnyshankman** [comment]
Issue is still present for me.

```
"@babel/core": "^7.2.2",
...
"babel-eslint": "^10.0.1",
...
"eslint": "^5.13.0",
```

> Until it get updated in ESLint proper

@existentialism I don't think this is the case anymore since ESLint 5 released but I'm not 100% sure. Seems like it from this discussion https://github.com/eslint/eslint/issues/9662
